File: src/backend/DonkeyWork.Chat.Api/Models/Prompt/GetActionPromptsItemModel.cs
// ------------------------------------------------------
// <copyright file="GetActionPromptsItemModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

using DonkeyWork.Chat.Common.Models.Prompt;

namespace DonkeyWork.Chat.Api.Models.Prompt;

/// <summary>
/// An action prompt item.
/// </summary>
public record GetActionPromptsItemModel
{
    /// <summary>
    /// Gets the id.
    /// </summary>
    public Guid Id { get; init; }

    /// <summary>
    /// Gets the Name.
    /// </summary>
    public string Name { get; init; } = string.Empty;

    /// <summary>
    /// Gets the description.
    /// </summary>
    public string Description { get; init; } = string.Empty;

    /// <summary>
    /// Gets the prompt variables.
    /// </summary>
    public Dictionary<string, PromptVariable> Variables { get; init; } = [];

    /// <summary>
    /// Gets the prompt messages.
    /// </summary>
    public List<PromptMessage> Messages { get; init; } = [];

    /// <summary>
    /// Gets the usage count.
    /// </summary>
    public int UsageCount { get; init; }

    /// <summary>
    /// Gets the Created At.
    /// </summary>
    public DateTimeOffset CreatedAt { get; init; }

    /// <summary>
    /// Gets the Updated At.
    /// </summary>
    public DateTimeOffset UpdatedAt { get; init; }
}

File: src/ui/src/models/api/Prompt/GetActionPromptsItemModel.ts
import { PromptMessage } from './Prompt';
import { ActionPromptMessageModel, ActionPromptVariable } from './UpsertActionPromptModel';

/**
 * Model for an action prompt item in the list view
 */
export interface ActionPromptItem {
  /**
   * The unique identifier for the action prompt
   */
  id: string;
  
  /**
   * The name of the action prompt
   */
  name: string;
  
  /**
   * The description of the action prompt
   */
  description: string;
  
  /**
   * The variables in the action prompt
   * Can be array (old format) or record (new format)
   */
  variables: Record<string, ActionPromptVariable> | any[];
  
  /**
   * The messages in the action prompt
   * This can be either PromptMessage[] or ActionPromptMessageModel[] depending on the context
   */
  messages: ActionPromptMessageModel[] | PromptMessage[];
  
  /**
   * The number of times the action prompt has been used
   */
  usageCount: number;
  
  /**
   * The timestamp when the action prompt was created
   */
  timestamp: Date;
  
  /**
   * API fields for backward compatibility
   */
  updatedAt?: string;
  createdAt?: string;
}

export type GetActionPromptsItemModel = ActionPromptItem;